firewood/src/merkle.rs

```diff
@@ -319,7 +306,7 @@ impl<S: ShaleStore<Node> + Send + Sync> Merkle<S> {
                     }
                     // insert path is greather than the path of the leaf
                     (Ordering::Greater, Some(n_value)) => {
-                        let leaf = Node::leaf(
+                        let leaf = Node::from_leaf(
```

Comment:
This is from PartialPath and Data, not from a LeafNode, which is what `from_leaf` implies. Maybe `Node::leaf_from_path_and_data`

Reply:
This is a mistake on my part, I should be passing in a `Leaf` there.